doc/introduction/pulse.rst

```diff
@@ -0,0 +1,339 @@
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+.. warning::
+    The order of the coefficients and operators matters. When initializing the
+    :class:`~.ParametrizedHamiltonian`, terms defined with fixed coefficients
+    have to come before parametrized terms to prevent discrepancy in the wire order.
+    When passing parameters, ensure that the order of the coefficient functions and
+    the order of the parameters match.
+
+Convenience functions for building a ParametrizedHamiltonian
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following convenience functions currently are available in PennyLane to assist in creating coefficient functions
+for a :class:`~.ParametrizedHamiltonian`:
+
+:html:`<div class="summary-table">`
+
+.. autosummary::
+    :nosignatures:
+
+    ~pennylane.pulse.constant
+    ~pennylane.pulse.pwc
+    ~pennylane.pulse.pwc_from_function
+    ~pennylane.pulse.rect
+
+:html:`</div>`
+
+Further examples
+^^^^^^^^^^^^^^^^
+
+A few additional examples of defining a :class:`~.ParametrizedHamiltonian` are provided here.
+
+Here we use :func:`~.rect` to create a parametrized coefficient that has a value of ``0`` outside the time interval
+``t=(1, 7)``, and is defined by ``jnp.polyval(p, t)`` within the interval:
+
+.. figure:: ../../doc/_static/pulse/rect_example.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+This can be used to create a :class:`~.ParametrizedHamiltonian` in the following way:
+
+.. code-block:: python
+
+    def f(p, t):
+        return jnp.polyval(p, t)
+    H = qml.pulse.rect(f, windows=[(1, 7)]) * qml.PauliX(0)
+
+    # inside the window
+    >>> H([3], t=2)
+    2.7278921604156494*(PauliX(wires=[0]))
+
+    # outside the window
+    H([3], t=0.5 )
+    >>> 0.0*(PauliX(wires=[0]))
+
+
+The function :func:`~.pwc` can be used to create a parametrized coefficient function that is piecewise constant
+within the interval ``t``, and 0 outside of it. When creating the callable, only the timespan is passed. The number
+of bins and values for the parameters are set when ``params`` is passed to the callable.
+
+.. figure:: ../../doc/_static/pulse/pwc_example.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+.. code-block:: python
+
+    from pennylane.pulse.convenience_functions import pwc
+
+    f1 = pwc(timespan=(2, 7))
+    H = f1 * qml.PauliX(0)
+
+    # passing pwc((2, 7)) an array evenly distributes the array values in the interval t=2 to t=7
+    H(params=[[1, 2, 3, 4, 5]], t=2.3)
+    >>> 1.0*(PauliX(wires=[0]))
+
+    # different time, same bin, same result
+    H(params=[[1, 2, 3, 4, 5]], t=2.5)
+    >>> 1.0*(PauliX(wires=[0]))
+
+    # next bin
+    H(params=[[1, 2, 3, 4, 5]], t=3.1)
+    >>> 2.0*(PauliX(wires=[0]))
+
+    # outside the window where the function is assigned non-zero values
```

Comment:
This reads a bit ambiguously, I think: 
- outside the window in which the function is assigned non-zero values, or
- outside the window, where the function is assigned non-zero values (the opposite in meaning).
Again, a very minor thing, as the intended meaning is clear from the context and code.